Ensure assert.try_to returns `None` on failure

pivotaljohn opened the pull request:
... otherwise, if the wrapped function returns `nil` for the result
the variable assigned that rvalue becomes _undefined_.

cppforlife commented on pkg/yttlibrary/assert.go:
```diff
@@ -51,6 +51,7 @@ func (b assertModule) TryTo(thread *starlark.Thread, f *starlark.Builtin, args s
 	retVal, err := starlark.Call(thread, lambda, nil, nil)
 	if err != nil {
+		retVal = starlark.None
 		errVal = starlark.String(err.Error())
```
what do you think of doing explicit return here

return starlark.Tuple{starlark.None, starlark.String(err.Error()}, nil

and below

return starlark.Tuple{retVal, starlark.None}, nil

pivotaljohn replied:
sure... will incorporate. More durably satisfies the "BuiltIn contract" (i.e. returning `starlark.Value`s) 👍🏻

cppforlife replied:
typically a good thing anyway, otherwise it's similar to thinking about fallthrough cases in switch statements (and those things are ripe for introducing bugs)

pivotaljohn replied:
done.

cari-lynn committed:
Ensure assert.try_to returns `None` on failure

... otherwise, if the wrapped function returns `nil` for the result
the variable assigned that rvalue becomes _undefined_.

Signed-off-by: John Ryan <jtigger@infosysengr.com>
